python/tskit/tables.py

```diff
@@ -2538,9 +2538,13 @@ def simplify(
             not referenced by mutations after simplification; new site IDs are
             allocated sequentially from zero. If False, the site table will not
             be altered in any way. (Default: True)
-        :param bool keep_unary: If True, any unary nodes (i.e. nodes with exactly
-            one child) that exist on the path from samples to root will be preserved
-            in the output. (Default: False)
+        :param str keep_unary: When to preserve unary nodes (i.e. nodes with
+            exactly one child) that exist on the path from samples to root. If
+            ``None`` all unary nodes are removed. If "all" all such nodes will
+            be preserved in the output. If "in_individuals", they are only preserved
+            if they are associated with an individual. (Note that the boolean values
+            ``True`` and ``False`` can be used as shortcuts for "all" and ``None``.)
+            (Default: None)
```

Comment:
This could lead to confusion; we should probably allow `"none"` and `"None"` as aliases for `None` here?

Oh, actually: how about we replace `None` with `"none"` here, and say the default is `"none"`, so that all the default values referred to in this text are strings?  (and then, of course, `None` will mean `"none"` below...)

Reply:
and I think that if we don't talk about `None` in the text we won't get users accidentally upper-casing `"None"`...

Reply:
Yes, that would be reasonable. At the moment it just errors out with strings other than "in_individuals". Happy to add "None" and "none" as aliases.

Reply:
How do we feel about `""` and `"False"`. I assume we ignore them, but this is all getting a bit horrible!

Reply:
Like this, @petrelharp?

```
      :param str keep_unary: When to preserve unary nodes (i.e. nodes with
            exactly one child) that exist on the path from samples to root. If
            "none" all unary nodes are removed. If "all" all such nodes will
            be preserved in the output. If "in_individuals", they are only preserved
            if they are associated with an individual. (Note that the boolean value
            ``True`` can be used as a shortcuts for "all" and ``False``,
            ``None``, or "None" as a shortcut for "none") (Default: "none")
```

Wow. 4 "none"s in a single line. 🤯

Reply:
I'm not sure about the API here - is it not clearer to have two mutually exclusive flags like the C API?

Reply:
Depends on what future uses might be. I mooted some at https://github.com/tskit-dev/tskit/issues/1120#issuecomment-770179852 and another at https://github.com/tskit-dev/tskit/pull/1156#issuecomment-772081032. Whatever is most obvious to the user, I guess.